swiftpkg/bzlmod/swift_deps.bzl

```
@@ -1,39 +1,154 @@
+
+    Args:
+        module_ctx: An instance of `module_ctx`.
+        from_package: The data from the `from_package` tag.
+        config_pkgs: The data from the `configure_package` tag.
+    """
+
+    # Read Package.resolved.
+    if from_package.resolved:
+        pkg_resolved = module_ctx.path(from_package.resolved)
+        resolved_pkg_json = module_ctx.read(pkg_resolved)
+        resolved_pkg_map = json.decode(resolved_pkg_json)
+    else:
+        resolved_pkg_map = dict()
+
+    # Get the package info.
+    pkg_swift = module_ctx.path(from_package.swift)
+    debug_path = module_ctx.path(".")
+    pkg_info = pkginfos.get(
+        module_ctx,
+        directory = str(pkg_swift.dirname),
+        debug_path = str(debug_path),
+        resolved_pkg_map = resolved_pkg_map,
+        collect_src_info = False,
+    )
+
+    # Collect all of the deps by identity
+    all_deps_by_id = {
+        dep.identity: dep
+        for dep in pkg_info.dependencies
+    }
+
+    # Collect the direct dep repo names
+    direct_dep_repo_names = []
+    direct_dep_pkg_infos = {}
+    for dep in pkg_info.dependencies:
+        bazel_repo_name = bazel_repo_names.from_identity(dep.identity)
+        direct_dep_repo_names.append(bazel_repo_name)
+        pkg_info_label = "@{}//:pkg_info.json".format(bazel_repo_name)
+        direct_dep_pkg_infos[pkg_info_label] = dep.identity
+
+    # Write info about the Swift deps that may be used by external tooling.
+    swift_deps_info_repo_name = "swift_deps_info"
+    swift_deps_info(
+        name = swift_deps_info_repo_name,
+        direct_dep_pkg_infos = direct_dep_pkg_infos,
+    )
+    direct_dep_repo_names.append(swift_deps_info_repo_name)
+
+    # Ensure that we add all of the transitive source control deps from the
+    # resolved file.
+    for pin_map in resolved_pkg_map.get("pins", []):
+        pin = pkginfos.new_pin_from_resolved_dep_map(pin_map)
+        dep = all_deps_by_id.get(pin.identity)
+        if dep != None:
+            continue
+        dep = pkginfos.new_dependency(
+            identity = pin.identity,
+            # Just use the identity for the name as we just need this to set
+            # up the repositories.
+            name = pin.identity,
+            source_control = pkginfos.new_source_control(pin = pin),
+        )
+        all_deps_by_id[dep.identity] = dep
+
+    # Find all of the local Swift packages and add them to the all_deps_by_id.
+    # A local Swift package can reference other local Swift packages. Hence, we
+    # need to check all of the transitive local Swift packages, not just the
+    # direct local packages. We do not need to worry about the source control
+    # deps because they are already listed in the Package.resolved.
+    to_process = [
+        dep
+        for dep in all_deps_by_id.values()
+        if dep.file_system
+    ]
+    for _ in range(100):
+        if len(to_process) == 0:
```

I believe idiomatic Starlark is this:
```suggestion
        if not to_process:
```